database/bigquery/bigquery.go

```diff
@@ -0,0 +1,363 @@
+package bigquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	nurl "net/url"
+	"strings"
+	"time"
+
+	bq "cloud.google.com/go/bigquery"
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/hashicorp/go-multierror"
+	uatomic "go.uber.org/atomic"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/iterator"
+	bqopt "google.golang.org/api/option"
+)
+
+func init() {
+	db := BigQuery{}
+	database.Register("bigquery", &db)
+}
+
+const (
+	DefaultMigrationsTableName = "schema_migrations"
+	DefaultQueryTimeout        = time.Duration(10 * time.Second)
+)
+
+var (
+	dbUnlocked = false
+	dbLocked   = true
+
+	ErrDBLocked         = errors.New("unable to obtain database lock")
+	ErrDBNotLocked      = errors.New("database already unlocked")
+	ErrMissingConfigArg = errors.New("config is a required argument")
```

Comment:
```suggestion
	ErrNoConfig = errors.New("no config")
```

just how they name this error type in most of the other drivers